source/base/polynomials_barycentric.cc

```diff
@@ -60,6 +61,8 @@ BarycentricPolynomials<dim>::get_fe_p_basis(const unsigned int degree)
 {
   std::vector<PolyType> polys;
 
+  const auto reference_cell = ReferenceCells::get_simplex<dim>();
+
   auto M = [](const unsigned int d) {
     return BarycentricPolynomial<dim, double>::monomial(d);
   };
```

Comment:
While you're there: This is a pretty pointless lambda function. It basically just provides a shorter name of the function called. I think that's not useful.